File: OfflineFetcher/app/src/main/java/de/jbamberger/offlinefetcher/di/jodel/JodelSubComponent.java
package de.jbamberger.offlinefetcher.di.jodel;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;
import de.jbamberger.offlinefetcher.ui.jodel.JodelActivity;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@JodelActivityScope
@Subcomponent(modules = {JodelModule.class/*, NetModule.class*/})
public interface JodelSubComponent extends AndroidInjector<JodelActivity> {

    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<JodelActivity> {
    }
}

File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/ui/jodel/feed/JodelSubComponent.kt
package de.jbamberger.offlineclient.ui.jodel.feed

import dagger.Subcomponent
import dagger.android.AndroidInjector
import de.jbamberger.offlineclient.ui.jodel.JodelActivity

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@JodelActivityScope
@Subcomponent(modules = [JodelModule::class])
interface JodelSubComponent : AndroidInjector<JodelActivity> {

    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<JodelActivity>()
}
